Troubleshooting BYOIDP (#4964)
mendix/docs: content/en/docs/developerportal/control-center/set-up-sso-byoidp.md

When you [add or remove an email domain from your company](/developerportal/control-center/#company) in Mendix, you will need to deactivate BYOIDP and reactivate it again to apply the changes for users logging on using the changed domains.

You cannot select which of your email domains are used for single sign-on. When you activate BYOIDP SSO, it will apply to all email domains which are registered to your company.

## 9 Troubleshooting{#troubleshooting}

If you have issues using BYOIDP, the following suggestions give an initial guide to resolving the issue.

### 9.1 Wrong Client Credentials

> Without proper exchange of client_id and client_secret between the IdP and Mendix platform, Mendix cannot authenticate at the /token endpoint and delegated login will fail. This can happen when the wrong client credentials are supplied.

Reviewer: ```suggestion
Without proper exchange of `client_id` and `client_secret` between the IdP and Mendix platform, Mendix cannot authenticate at the /token endpoint and delegated login will fail. This can happen when the wrong client credentials are supplied.
```

### 9.2 Wrong Authentication Method

If the wrong authentication method is configured for Mendix platform as a client to the customer’s IdP login will fail.

Mendix supports two client authentication methods: `client_secret_post` (client credentials in the payload) or `client_secret_basic` (basic authentication; credentials in http authorization header). If the IdP indicates support for both methods at the well-known endpoint, Mendix will use `client_secret_post`. If the client configuration at the customer’s IdP for the Mendix platform sets a different client authentication method, the IdP may reject Mendix to authenticate itself as a client to the /token endpoint and delegated login will fail.

### 9.3 Incorrect Conditional Access Policies

> Studio Pro uses an embedded browser for login. The customer’s conditional access policies in AzureAD may block Studio Pro this. For example, customers using Microsoft’s Intune for MDM/MAM will hit this limitation.

Reviewer: ```suggestion
Studio Pro uses an embedded browser for login. The customer’s conditional access policies in AzureAD may block Studio Pro for this. For example, customers using Microsoft’s Intune for MDM/MAM will hit this limitation.
```

> Mendix has created an enhancement of Studio Pro that uses system browser to overcome this limitation. This feature is in ‘private beta’ and can be provided on request.

Reviewer: ```suggestion
Mendix has created an enhancement of Studio Pro that uses system browser to overcome this limitation. This feature is in Private Beta and can be provided on request.
```

### 9.4 IdP Does Not Allow Untrusted Devices

Studio Pro uses an embedded browser for login. Login via embedded browser may not work when customer’s IdP is allowing only for trusted devices and doesn’t recognize the embedded browser.

> Mendix has created an enhancement of Studio Pro that uses system browser to overcome this limitation. This feature is in ‘private beta’ and can be provided on request.

Reviewer: ```suggestion
Mendix has created an enhancement of Studio Pro that uses system browser to overcome this limitation. This feature is in Private Beta and can be provided on request.
```

### 9.5 IdP Does Not Have JWKS Endpoint

> If the IDP’s well-known endpoint doesn’t have a URL for the JWKS endpoint, Mendix cannot validate the signature on the received ID-token and the delegated authentication fails.

Reviewer: ```suggestion
If the IDP’s well-known endpoint does not have a URL for the JWKS endpoint, Mendix cannot validate the signature on the received ID-token and the delegated authentication fails.
```
